data/src/main/java/com/linkedin/data/schema/resolver/AbstractMultiFormatDataSchemaResolver.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2015 Coursera Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.data.schema.resolver;
+
+import com.linkedin.data.schema.DataSchemaLocation;
+import com.linkedin.data.schema.DataSchemaParserFactory;
+import com.linkedin.data.schema.DataSchemaResolver;
+import com.linkedin.data.schema.Name;
+import com.linkedin.data.schema.NamedDataSchema;
+import com.linkedin.data.schema.SchemaParser;
+import com.linkedin.data.schema.SchemaParserFactory;
+import com.linkedin.data.schema.grammar.PdlSchemaParser;
+import com.linkedin.data.schema.grammar.PdlSchemaParserFactory;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+
+/**
+ * Combines multiple file format specific resolvers (and respective file format specific parsers)
+ * into a single resolver.
+ *
+ * E.g. a resolver for the ".pdsc" file format and the ".pdl" file format, each with their
+ * own file format specific parsers, can be combined into a single resolver able to look up
+ * schemas of either file format.
+ */
+public class AbstractMultiFormatDataSchemaResolver implements DataSchemaResolver
```

Comment:
How is this different from current `MultiFormatDataSchemaResolver` (https://github.com/linkedin/rest.li/blob/master/data/src/main/java/com/linkedin/data/schema/resolver/MultiFormatDataSchemaResolver.java)?

Reply:
Based on my understanding, he wanted to add similar logic to `ClasspathResourceDataSchemaResolver`, so it makes more sense to add the shared logic in a common abstract class.

Reply:
Correct, refactored the existing resolver to extract the common logic to abstract class. Forgot to make it abstract, will do it.

The existing multi-format resolver is the file based one.

Reply:
Never mind, I found out when I read through to the end.